src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -93,20 +93,26 @@ public static class AttributeMatcher
             var presentAttributes = new List<string>();
             var missingAttributes = new List<string>();
 
-            foreach (var attribute in requiredAttributes)
+            foreach (var requiredAttribute in requiredAttributes)
             {
-                if (!context.AllAttributes.ContainsName(attribute) ||
-                    context.AllAttributes[attribute] == null ||
-                    (context.AllAttributes[attribute].Value is string &&
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute].Value as string)))
+                IReadOnlyTagHelperAttribute attribute;
+                if (!context.AllAttributes.TryGetAttribute(requiredAttribute, out attribute) ||
+                    attribute == null)
```

Comment:
Put another way, none of the methods in `TagHelperExecutionContext` can add `null` to the source for that class' `AttAttributes` collection. And that's the source for this collection.